model/src/main/proto/spine_examples/shareaware/investment/commands.proto

```diff
@@ -49,3 +49,45 @@ message AddShares {
     // The number of shares to add.
     int32 quantity = 3 [(required) = true];
 }
+
+// A command to reserve shares for their sale.
+message ReserveShares {
+
+    // The ID of the investment where to reserve shares.
+    InvestmentId investment = 1;
+
+    // The ID of the process that is about to reserve shares.
+    SaleId process = 2 [(required) = true];
+
+    // The quantity of shares to reserve.
+    int32 quantity = 3 [(required) = true];
+}
+
+// A command to close the shares reservation.
```

Comment:
Let's then call it `Complete...`. "Close" sounds like a "bad" outcome, not too much different from "Cancel" which you already have.